### PR TITLE
Applications: Migrate to Core::Directory::for_each_entry()

### PR DESCRIPTION
**Pull request by AtkinsSJ**
A lot of this is yaks to get SpaceAnalyzer ported.

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
Rebased and fixed conflicts.

File: Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp [hunk 1]
```diff
@@ -73,11 +72,11 @@ ErrorOr<void> ThemesSettingsWidget::setup_interface()
...
-    Core::DirIterator iterator("/res/color-schemes", Core::DirIterator::SkipParentAndBaseDir);
-    while (iterator.has_next()) {
-        auto path = iterator.next_path();
-        TRY(m_color_scheme_names.try_append(TRY(String::from_deprecated_string(path.replace(".ini"sv, ""sv, ReplaceMode::FirstOnly)))));
-    }
+    Core::Directory::for_each_entry("/res/color-schemes"sv, Core::DirIterator::SkipParentAndBaseDir, [&](auto& entry, auto&) -> ErrorOr<IterationDecision> {
+        LexicalPath path { entry.name };
+        TRY(m_color_scheme_names.try_append(TRY(String::from_utf8(path.title()))));
+        return IterationDecision::Continue;
+    }).release_value_but_fixme_should_propagate_errors();
```
Comment (supercomputer7) [inline Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp:79]: Is there a reason to not just propagate this with `TRY()`? `ThemesSettingsWidget::setup_interface()` returns `ErrorOr<void>` anyways.

File: Userland/Applications/Terminal/main.cpp [hunk 2]
```diff
@@ -355,12 +355,11 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
     auto shell_child_process_count = [&] {
-        Core::DirIterator iterator(DeprecatedString::formatted("/proc/{}/children", shell_pid), Core::DirIterator::Flags::SkipParentAndBaseDir);
         int background_process_count = 0;
-        while (iterator.has_next()) {
+        Core::Directory::for_each_entry(DeprecatedString::formatted("/proc/{}/children", shell_pid), Core::DirIterator::Flags::SkipParentAndBaseDir, [&](auto&, auto&) {
             ++background_process_count;
-            (void)iterator.next_path();
-        }
+            return IterationDecision::Continue;
+        }).release_value_but_fixme_should_propagate_errors();
```
Comment (supercomputer7) [inline Userland/Applications/Terminal/main.cpp:362]: Is there a reason to not just propagate this with TRY()?

**Comment by supercomputer7** [comment]
Besides my 2 comments on this, this looks OK to me :)

File: Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp]: Thanks, I missed this!

**Comment by AtkinsSJ** [comment]
Changes: Propagate the error in ThemesSettingsWidget.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**